src/access.d

```diff
@@ -538,3 +519,78 @@ extern (C++) bool symbolIsVisible(Scope *sc, Dsymbol s)
     case PROTpublic, PROTexport: return true;
     }
 }
+
+/**
+ * Use the most visible overload to check visibility. Later perform an access
+ * check on the resolved overload.  This function is similar to overloadApply,
+ * but doesn't recurse nor resolve aliases because protection/visibility is an
+ * attribute of the alias not the aliasee.
+ */
+private Dsymbol mostVisibleOverload(Dsymbol s)
```

Comment:
I plan some refactoring to merge the new visibility mechanism with the old one (limiting visibility of private imports). When this is done, the signature would become `bool hasVisibleOverload(Dsymbol s, Scope* sc, Prot visibility)`, but right now we need to dig out an actual symbol to check.